lessonbuilder/tool/src/java/org/sakaiproject/lessonbuildertool/ccexport/CCExport.java

```diff
@@ -144,6 +156,70 @@ public void doExport(HttpServletResponse response, String siteId, String version
             outputAllForums(ccConfig, out);
             outputAllBlti(ccConfig, out);
             outputAllTexts(ccConfig, out);
+
+            // common/archive-impl/impl2/src/java/org/sakaiproject/archive/impl/SiteArchiver.java
+
+            // this is the folder we are writing files to
+            String archiveStorage = storagePath + siteId + "-archive/";
+
+            // create the directory for the archive
+            File dir = new File(archiveStorage + siteId + "-archive/");
+
+            // clear the directory (if site already archived) so resources are not duplicated
+            try {
+                FileUtils.deleteDirectory(dir);
+            } catch (IOException e) {
+                log.warn("Could not clear existing archive: {}: {}", dir, e.toString());
+            }
+
+            // collect all the attachments we need
+            List attachments = entityManager.newReferenceList();
+            List<String> exportedLabels = new ArrayList<String>(); // LTI registers two providers
+            Collection<EntityProducer> producers = entityManager.getEntityProducers();
+            for (EntityProducer producer : producers) {
+                if (producer == null) continue;
+
+                final String serviceName = producer.getClass().getCanonicalName();
+                final String serviceLabel = producer.getLabel();
+
+                if (!producer.willArchiveMerge()) continue;
+                if ( exportedLabels.contains(serviceLabel) ) continue;
+
+                Document doc = Xml.createDocument();
+                Stack stack = new Stack();
+                Element root = doc.createElement("archive");
+                doc.appendChild(root);
+                root.setAttribute("source", siteId);
+
+                stack.push(root);
+
+                try {
+                    String archive = producer.archive(siteId, doc, stack, archiveStorage, attachments);
+                    String xml = Xml.writeDocumentToString(doc);
+
+                    String zipId = "archive-"+serviceLabel;
+                    String zipName = "sakai_archive/"+serviceLabel+".xml";
+                    ZipEntry archiveEntry = new ZipEntry(zipName);
+                    out.putNextEntry(archiveEntry);
+                    out.print(xml);
+
+                    String resourceId = ccConfig.getResourceId();
+                    CCResourceItem res = new CCResourceItem(zipId, resourceId, zipName, null, null, null);
+                    ccConfig.getArchiveMap().put(zipId, res);
+
+                    exportedLabels.add(serviceLabel);
+
+                } catch (java.lang.Throwable t) {
+                    String archiveError = "Error archiving "+serviceLabel+" "+serviceName+" "+t.toString();
+                    log.error(archiveError);
+                    t.printStackTrace();
+                    ccConfig.getResults().add(archiveError);
+                }
+
+// TODO: Handle attachments...
+System.out.println("attachments="+attachments);
```

Comment:
remove

Reply:
Yup - is draft with TODOs.  Very incomplete.